### ISSUE TITLE
Compact Cluster Support

### ISSUE THREAD

**Issue by christianh814** [RedHatOfficial/ocp4-helpernode#108]
Support for "Compact" clusters (3 nodes that are both master and worker) is coming in 4.5

Currently, this playbook doesn't support such design. the HAproxy config must be edited to account for the router running on the masters.

Other things to consider is DNS and DHCP configurations.

**Comment by mransonwang** [comment]
@christianh814 I'm considering install ocp 4.5 with helper node, would you give some advice on how to modify DNS, HAproxy, DHCP manually, so the ocp 4.5 three nodes cluster could be installed smoothly, many thanks.

**Comment by christianh814** [comment]
@mransonwang 

For now I would use "dummy" values for the workers. For example in your `vars.yaml` file, it can look like this.

```yaml
---
disk: vda
helper:
  name: "helper"
  ipaddr: "192.168.7.77"
dns:
  domain: "example.com"
  clusterid: "ocp4"
  forwarder1: "8.8.8.8"
  forwarder2: "8.8.4.4"
dhcp:
  router: "192.168.7.1"
  bcast: "192.168.7.255"
  netmask: "255.255.255.0"
  poolstart: "192.168.7.10"
  poolend: "192.168.7.30"
  ipid: "192.168.7.0"
  netmaskid: "255.255.255.0"
bootstrap:
  name: "bootstrap"
  ipaddr: "192.168.7.20"
  macaddr: "52:54:00:60:72:67"
masters:
  - name: "master0"
    ipaddr: "192.168.7.21"
    macaddr: "52:54:00:e7:9d:67"
  - name: "master1"
    ipaddr: "192.168.7.22"
    macaddr: "52:54:00:80:16:23"
  - name: "master2"
    ipaddr: "192.168.7.23"
    macaddr: "52:54:00:d5:1c:39"
workers:
  - name: "worker0-dummy"
    ipaddr: "192.168.7.11"
    macaddr: "52:54:00:f4:26:a1"
  - name: "worker1-dummy"
    ipaddr: "192.168.7.12"
    macaddr: "52:54:00:82:90:00"
```

Run the playbook with these "dummy" entries.



Then, after the playbook is ran, edit the following files

* `/var/named/zonefile.db` Edit this file to remove all the `worker` entries
* `/var/named/reverse.db` Edit this file to remove all the `worker` entries
* `/etc/dhcp/dhcpd.conf` Edit this file to remove all the `worker` entries
* `/var/lib/tftpboot/pxelinux.cfg/` You need to `cd` into this directory and remove the files that has the "dummy" MAC addresses
* `/etc/haproxy/haproxy.cfg` Edit this file and change the IP for the worker addresses to point to the master address

Then restart all services

```
systemctl restart named dhcpd tftp haproxy
```

Check the services

```
helpernodecheck services
```
:warning: Note that `tftp` might appear "dead" (this is okay)

After creating the manifests, **_SKIP_** this step

```shell
$ sed -i 's/mastersSchedulable: true/mastersSchedulable: false/g' manifests/cluster-scheduler-02-config.yml # DON'T DO THIS STEP
```

Can you test this out? If it works for you I'll add it to the documentation as a stopgap until we get a 3 node "compact" cluster supported.

**Comment by mransonwang** [comment]
@christianh814 Yes, will try it

**Comment by mransonwang** [comment]
@christianh814 Just verified the 3 masters OCP installation, the installation looked fine. How ever, due to my environment resources limitation, some pods status were Preempting, I think it shoud not related with helper node

**Comment by christianh814** [comment]
Closed by #127
